File: projects/grind-75/grind-75-python/linked_list/reverse_linked_list.py
class Solution:
    @classmethod
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        # using curr
        prev = None
        while head:
            current = head
            head = head.next
            current.next = prev
            prev = current
        return prev
        # using nxt
        # prev = None
        # while head:
        #     nxt = head.next
        #     head.next = prev
        #     prev = head
        #     head = nxt
        # return prev

File: projects/grind-75/typescript/linked-list/reverse-linked-list.ts
import { ListNode } from './common';

export default function reverseList(head: ListNode | null): ListNode | null {
  let prev: ListNode | null = null;

  while (head) {
    const curr = head;
    head = curr.next;
    curr.next = prev;
    prev = curr;
  }

  // while (head){
  //     const next = head.next
  //     head.next = prev
  //     prev = head
  //     head = next
  // }

  return prev;
}
